lib/spack/spack/compilers/msvc.py

```diff
@@ -122,7 +122,15 @@ def platform_toolset_ver(self):
     @property
     def cl_version(self):
         """Cl toolset version"""
-        return spack.compiler.get_compiler_version_output(self.cc)
+        return Version(
+            re.search(
+                Msvc.version_regex, spack.compiler.get_compiler_version_output(self.cc, "")
+            ).group(1)
+        )
+
+    @property
+    def vs_root(self):
+        return os.path.abspath(os.path.join(self.cc, "../../../../../../../.."))
```

Comment:
This looks fragile, is there some way to query the compiler for the root? If not, can a docstring mention the structure?

Reply:
We rely on almost exactly the same logic in WindowsOS to find the VCVARS script. One of the nice things about MSVC is its relatively consistent structure. so while this looks incredibly fragile. it should be solid, and if it does break, our VCVARS detection is also broken so nothing will compile anyway.

Reply:
I'll add a docstring.